tests/rbd_mirror/test_delayed_replication_handler.py

```diff
@@ -0,0 +1,190 @@
+       delay that there are no mirroring activity between clusters
+
+    2. polarion-id: CEPH-11490- Delayed Replication-Failover with delay set.
+
+       If skip_initial_config is set, record initial md5sum value of images,
+       perform force failover operation after some io operation.
+       Check md5sum of images and comapre with initial value.
+
+    """
+    log.info("Starting execution for Delayed replication scenario")
+
+    config = kw.get("config")
+
+    pool_image_config = {
+        "rep_pool_config": {
+            "mode": "pool",
+            "mirrormode": "journal",
+            "image": "rbd_image_rep",
+            "pool": "rbd_pool_delayed_replication_handler",
+            "io_total": "100M",
+        },
+        "ec_pool_config": {
+            "pool": "rbd_pool_delayed_replication_handler",
+            "image": "rbd_image_ec",
+            "mode": "image",
+            "data_pool": "data_pool",
+            "mirrormode": "journal",
+            "io_total": "100M",
+        },
+    }
+    kw["config"].update(pool_image_config)
+    imagespecs = [
+        f'{kw["config"]["rep_pool_config"]["pool"]}/{kw["config"]["rep_pool_config"]["image"]}',
+        f'{kw["config"]["ec_pool_config"]["pool"]}/{kw["config"]["ec_pool_config"]["image"]}',
+    ]
+    # TBD: increase delay to 600s
+    delay = config.get("delay", 600)
+
+    mirror1, mirror2 = [
+        rbdmirror.RbdMirror(cluster, config)
+        for cluster in kw.get("ceph_cluster_dict").values()
+    ]
+
+    rbd1, rbd2 = [
+        Rbd(**kw, req_cname=cluster_name)
+        for cluster_name in kw.get("ceph_cluster_dict").keys()
+    ]
+
+    def export_and_get_md5sum():
+        md5s = {}
+        with parallel() as p:
+            p.spawn(mirror2.export_image, imagespec=imagespecs[0], path="export0")
+            p.spawn(mirror2.export_image, imagespec=imagespecs[1], path="export1")
+        try:
+            with parallel() as p:
+                p.spawn(
+                    mirror2.exec_cmd, cmd="md5sum export0", output=True, ceph_args=False
+                )
+                p.spawn(
+                    mirror2.exec_cmd, cmd="md5sum export1", output=True, ceph_args=False
+                )
+
+                for result in p:
+                    md5s.update(
+                        {f"{result.split('  ')[1][:-1]}": f"{result.split('  ')[0]}"}
+                    )
+            return md5s
+        finally:
+            with parallel() as p:
+                p.spawn(
+                    mirror2.exec_cmd, cmd="rm export0", output=True, ceph_args=False
+                )
+                p.spawn(
+                    mirror2.exec_cmd, cmd="rm export1", output=True, ceph_args=False
+                )
+
+    def get_initial_md5sums():
+        log.debug(
+            "Waiting for reply to complete and fetching initial md5sum of image exports"
+        )
+        with parallel() as p:
+            p.spawn(mirror2.wait_for_replay_complete, imagespecs[0])
```

Comment:
use iterator

Reply:
Done.

Modified it in few more places also.